src/net/java/sip/communicator/impl/protocol/jabber/ProtocolProviderServiceJabberImpl.java

```diff
@@ -1114,8 +1114,40 @@ private ConnectState connectAndLogin(
         confConn.setXmppDomain(serviceName);
         if (isBosh)
         {
-            confConn.setHost(boshURL);
-            // FIXME use setHostAddress, use proxy
+            BOSHConfiguration.Builder boshConfConnBuilder =
+                (BOSHConfiguration.Builder)confConn;
+
+            try
+            {
+                URI boshURI = new URI(boshURL);
+                boolean useHttps = boshURI.getScheme().equals("https");
+
+                int port = boshURI.getPort();
+                if (port == -1)
+                {
+                    port = useHttps ? 443 : 80;
+                }
+
+                String file = boshURI.getPath();
+                String query = boshURI.getQuery();
+                if (!StringUtils.isNullOrEmpty())
```

Comment:
Missing argument? ;-)
I'm surprised that this even compiles.

Reply:
Shit, sorry, hum...

Reply:
Indeed it was compiling, because of isNotEmpty(CharSequence... css).
Fixed now.